source/defclass-star.lisp

```diff
@@ -522,3 +522,58 @@ See `define-class' for more details."
      `(define-condition ,name ,supers
         ,processed-slots
         ,@clean-options))))
+
+(defmacro make-instance* (class &rest rest)
+  "Convenience macro on top of `make-instance'.
+
+Conveniences:
+- First argument can be a list of symbols, which is automatically
+  converted into a keyword+symbol pairs.
+- The last argument can be an arbitrary list-producing form, which
+  will be used as the last argument to (apply #'make-instance ...).
+
+Otherwise, the behavior is equivalent to `make-instance'.
+
+Examples:
+
+Shortcut args:
+\(make-instance* 'class (a b c))
+=>
+\(make-instance 'class :a a :b b :c c)
+
+Last form for `apply':
+\(make-instance* 'class :a a :b b (when something (list a)))
+=>
+\(apply #'make-instance 'class :a a :b b (when something (list a)))
+
+Shortcut arguments, regular arguments, and form for `apply':
+\(make-instance* 'class (a b c) :x x (when something (list a)))
+=>
+\(apply #'make-instance 'class :a a :b b :c c :x x (when something (list a)))"
+  (let* ((maybe-shortcut-arguments (first rest))
+         (shortcuts-p (and (listp maybe-shortcut-arguments)
+                           (every #'symbolp maybe-shortcut-arguments)
+                           (not (fboundp (first maybe-shortcut-arguments)))))
+         (shortcut-arguments (if shortcuts-p
+                                 (first rest)
+                                 nil))
+         (arguments (if shortcuts-p
+                        (rest rest)
+                        rest))
+         (last-appendable-form (if (oddp (length arguments))
+                                   (car (last arguments))
```

Comment:
`car` -> `first`.

Reply:
Here the use is ambiguous, but the idea is to reproduce `alex:lastcar`. `last` explicitly returns a cons cell, not a list. Handling cons cells (as per SICL style guide) should happen with `car`/`cdr`, so that `first` does not trick us into thinking that the value is a valid list.

But I can change it to `first` too.